api/openapi.yaml

```diff
@@ -738,335 +738,478 @@ paths:
+      operationId: getEntitlementHistory
+      summary: Get the balance history of a specific entitlement.
       description: |
-        Gets the grant for a ledger by ID.
+        Get the balance history of a specific entitlement.
+
+        If windowing is specified then groups the history by the window size starting from "to". The windows are inclusive at their start and exclusive at their end. The last chunk may be smaller than the window size and is inclusive at both ends.
+
+        If windowing is not specified then the returns a continous history.
       tags:
         - Entitlements (Experimental)
       parameters:
-        - $ref: "#/components/parameters/ledgerID"
-        - $ref: "#/components/parameters/ledgerGrantID"
+        - $ref: "#/components/parameters/subjectIdOrKey"
+        - $ref: "#/components/parameters/entitlementId"
+        - name: from
+          in: query
+          required: true
+          description: |
+            Start of time range to query entitlement: date-time in RFC 3339 format.
+            Gets truncated to the granularity of the underlying meter.
+          schema:
+            type: string
+            format: date-time
+            example: "2023-01-01T00:00:00Z"
+        - name: to
+          in: query
+          required: false
+          description: |
+            End of time range to query entitlement: date-time in RFC 3339 format. Defaults to now.
+            If not now then gets truncated to the granularity of the underlying meter.
+          schema:
+            type: string
+            format: date-time
+            example: "2023-01-01T00:00:00Z"
+        - name: windowSize
+          in: query
+          required: false
+          description: |
+            Size of the time window to group the history by. Cannot be shorter than meter granularity.
+          schema:
+            type: string
+            enum:
+              - MINUTE
+              - HOUR
+              - DAY
       responses:
         "200":
-          description: Ledger grant found.
+          description: |
+            Entitlement balance history. If windowsize is specified then the history is grouped by the window size.
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/LedgerGrantResponse"
+                oneOf:
+                  - $ref: "#/components/schemas/GrantBurnDownHistory"
+                  - $ref: "#/components/schemas/WindowedBalanceHistory"
+        "400":
+          $ref: "#/components/responses/BadRequestProblemResponse"
         "401":
           $ref: "#/components/responses/UnauthorizedProblemResponse"
         "404":
           $ref: "#/components/responses/NotFoundProblemResponse"
         default:
           $ref: "#/components/responses/UnexpectedProblemResponse"
-    delete:
-      operationId: voidLedgerGrant
-      summary: Void ledger grant
+  /api/v1/subjects/{subjectIdOrKey}/entitlements/{entitlementId}/reset:
```

Comment:
This is updating the `periodAnchor`. Wondering if this need to be a separate endpoint or it's an edit on the entitlement